src/go/web/handlers.go

```diff
@@ -3332,6 +3338,214 @@ func GetError(w http.ResponseWriter, r *http.Request) {
 	w.Write(body)
 }
 
+// POST /experiments/{exp}/vms/{name}/mount
+func MountVM(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+
+	role := r.Context().Value("role").(rbac.Role)
+	if !role.Allowed("vms/mount", "patch") {
+		http.Error(w, "forbidden", http.StatusForbidden)
+		return
+	}
+
+	mountsMu.Lock()
+	defer mountsMu.Unlock()
+	_, err := mm.ExecC2Command(mm.C2NS(vars["exp"]), mm.C2VM(vars["name"]), mm.C2Mount(), mm.C2IDClientsByUUID())
+
+	// if already mounted, that's fine
+	if err != nil && !strings.Contains(err.Error(), "already connected") {
+		log.Error(err.Error())
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	activeMounts[mm.GetLocalMountPath(vars["exp"], vars["name"])] = struct{}{}
+	
+	w.WriteHeader(http.StatusOK)
+}
+
+// POST /experiments/{exp}/vms/{name}/unmount
+func UnmountVM(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+
+	role := r.Context().Value("role").(rbac.Role)
+	if !role.Allowed("vms/mount", "patch") {
+		http.Error(w, "forbidden", http.StatusForbidden)
+		return
+	}
+
+	mountsMu.Lock()
+	defer mountsMu.Unlock()
+	_, err := mm.ExecC2Command(mm.C2NS(vars["exp"]), mm.C2VM(vars["name"]), mm.C2Unmount(), mm.C2IDClientsByUUID())
+
+	if err != nil {
+		log.Error(err.Error())
+		http.Error(w, fmt.Sprintf("Error unmounting: %s", err), http.StatusInternalServerError)
+		return
+	}
+
+	delete(activeMounts, mm.GetLocalMountPath(vars["exp"], vars["name"]))
+
+	w.WriteHeader(http.StatusOK)
+}
+
+// GET /experiments/{exp}/vms/{name}/mount/files?path=
+func GetMountFiles(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	basePath := mm.GetLocalMountPath(vars["exp"], vars["name"])
+
+	role := r.Context().Value("role").(rbac.Role)
+	if !role.Allowed("vms/mount", "patch") {
```

Comment:
Should use `list` instead of `patch` here since we're getting a list of mounted files, and should pass `{exp_name}/{vm_name}` to `role.Allowed` as well to support per-VM access control for mounts.